apps/prairielearn/src/tests/questionSharing.test.ts

```diff
@@ -0,0 +1,252 @@
+import ERR = require('async-stacktrace');
+import { assert } from 'chai';
+import { step } from 'mocha-steps';
+import { config } from '../lib/config';
+import fetch from 'node-fetch';
+import helperClient = require('./helperClient');
+import helperServer = require('./helperServer');
+import sqldb = require('@prairielearn/postgres');
+const sql = sqldb.loadSqlEquiv(__filename);
+import { EXAMPLE_COURSE_PATH, TEST_COURSE_PATH } from '../lib/paths';
+
+import syncFromDisk = require('../sync/syncFromDisk');
+
+import { makeMockLogger } from './mockLogger';
+const { logger } = makeMockLogger();
+
+const siteUrl = 'http://localhost:' + config.serverPort;
+const baseUrl = siteUrl + '/pl';
+
+const UUID_REGEXP = /[0-9a-fA-F]{8}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{12}/;
+
+const testCourseId = 1;
+const testCourseSharingName = 'test-course';
+const exampleCourseId = 2;
+const exampleCourseSharingName = 'example-course';
+const sharingSetName = 'share-set-example';
+
+function sharingPageUrl(courseId) {
+  return `${baseUrl}/course/${courseId}/course_admin/sharing`;
+}
+
+async function setSharingName(courseId, name) {
+  const sharingUrl = sharingPageUrl(courseId);
+  const response = await helperClient.fetchCheerio(sharingUrl);
+
+  const token = response.$('#test_csrf_token').text();
+  return await fetch(sharingUrl, {
+    method: 'POST',
+    headers: { 'Content-type': 'application/x-www-form-urlencoded' },
+    body: new URLSearchParams({
+      __action: 'unsafe_choose_sharing_name',
+      __csrf_token: token,
+      course_sharing_name: name,
+    }).toString(),
```

Comment:
Nit, it shouldn't be necessary to explicitly serialize to a string.

```suggestion
    }),
```

Reply:
fixed all instances